src/ape/api/networks.py

```diff
@@ -246,10 +246,13 @@ def use_provider(
             provider_name = provider_name.split(":")[0]
 
         if provider_name in self.providers:
-            return ProviderContextManager(
-                self.ecosystem.network_manager,
-                self.providers[provider_name](provider_settings=provider_settings),
-            )
+            try:
+                return ProviderContextManager(
+                    self.ecosystem.network_manager,
+                    self.providers[provider_name](provider_settings=provider_settings),
```

Comment:
I wrapped the `return` statement, since I wasn't sure how to wrap just the one argument.

Reply:
More like this (pull out the statement that needs the try...except)
```python
try:
    provider = self.providers[provider_name](provider_settings=provider_settings)
 ...
 
return ProviderContextManager(
    ...,
    provider,
    ...,
)
```